app/models/post.rb

```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def user_counter_updater
+    total_posts = User.find_by(User_id).posts_counter
+    User.find_by(User_id).update(posts_counter: total_posts + 1)
```

Comment:
- [ ] [OPTIONAL] To make this a bit easier you might want to use the below code `user.increment!(:posts_counter)`